volatility/framework/plugins/linux/lsmod.py

```diff
@@ -44,6 +47,12 @@ def list_modules(cls, context: interfaces.context.ContextInterface, layer_name:
 
         vmlinux = contexts.Module(context, vmlinux_symbols, layer_name, 0)
 
+        try:
+            vmlinux.get_type("module")
+        except exceptions.SymbolError:
+            vollog.debug("The required symbol 'module' is not present in symbol table. Please check that kernel modules are enabled for the system under analysis.") 
+            return
```

Comment:
So as discussed, I'd prefer not to catch the exception inside this method, but leave it to callers to deal with.

Reply:
If we catch around the for loop in the `_generator` method, then that means callers can distinguish between error states and empty lists.